### ISSUE TITLE
[SOLVED] Theoretical Ground for the Warmup Epochs

### ISSUE THREAD

**Issue by LucasVandroux** [experiencor/basic-yolo-keras#164]
Hello @experiencor,

I was wondering how you came up with the warmup process and the number of epochs necessary to make it useful. I understand from the `README.md` that those 3 to 5 epochs are here to help the network get familiar with the dataset before the training. Moreover, the main difference between the training epochs and the warmup ones is that the size of the boxes is forced to match the size of the anchors.

However, how did you come up with that idea? Is there a paper about it? Did you try different intuitions and it was the one working the best?

I am just interested to know why it works that way 🤓 and I couldn't find much information about it. 

Thanks a lot for your time.

**Comment by experiencor** [comment]
@LucasVandroux I did not come up with that idea on my own :smile:. It's not mentioned in the YOLO paper but it's in the official implementation (https://github.com/pjreddie/darknet/).

My opinion is that like many other techniques in deep learning people try different ideas based on intuition. Some ideas just happen to result in a better performance. Examples are ReLU or BatchNorm. So the results (and some hand-waving math arguments) would justify the ideas rather rigorous theoretical treatments.

**Comment by LucasVandroux** [comment]
@experiencor thanks, for the answer. Maybe one day, someone will come up with a rigorous demonstration. For the moment, the good results are good enough for me 😜
